packages/next/src/lib/typescript/writeAppTypeDeclarations.ts

```diff
@@ -55,6 +57,10 @@ export async function writeAppTypeDeclarations({
     )
   }
 
+  directives.push(
+    `/// <reference path="./${distDir ?? '.next'}/types/routes.d.ts" />`
```

Comment:
```suggestion
    `/// <reference path="./${path.posix.join(distDir ?? '.next', '/types/routes.d.ts')}" />`
```
We might wanna use join here since it's taking arbitrary distDir value.

Reply:
Is `distDir` a system path or posix path? If it's a system path, we're missing a transform to a posix path which `path.posix.join` won't do
```suggestion
    `/// <reference path="./${path.posix.join(distDir.replaceAll(path.win32.sep, path.posix.sep) ?? '.next', 'types/routes.d.ts')}" />`
```

Reply:
Good point, just fixed this.